openshift-kube-apiserver/admission/customresourcevalidation/dns/validate_dns.go

```diff
@@ -142,3 +143,32 @@ func validateTolerations(versionedTolerations []corev1.Toleration, fldPath *fiel
 	allErrors = append(allErrors, apivalidation.ValidateTolerations(unversionedTolerations, fldPath)...)
 	return allErrors
 }
+
+// validateUpstreamResolvers validates configuration of the Upstream objects when TLS is configured.
+func validateUpstreamResolvers(upstreamResolvers operatorv1.UpstreamResolvers, fieldPath *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	tc := operatorv1.DNSTransportConfig{}
+
+	// TransportConfig is empty so mixed Upstream types are allowed.
+	if upstreamResolvers.TransportConfig == tc {
+		return errs
+	}
+
+	// Transport is empty or Cleartext so mixed Upstream types are allowed.
+	if upstreamResolvers.TransportConfig.Transport == "" || upstreamResolvers.TransportConfig.Transport == operatorv1.CleartextTransport {
+		return errs
+	}
+
+	// Transport is TLS so no mixed Upstream types are allowed.
+	if upstreamResolvers.TransportConfig.Transport == operatorv1.TLSTransport {
```

Comment:
It would make sense to use `switch` here:
```suggestion
	switch upstreamResolvers.TransportConfig.Transport {
	case "", operatorv1.CleartextTransport:
		// Transport is empty or Cleartext so mixed Upstream types are allowed.
		return errs
	case operatorv1.TLSTransport:
		// Transport is TLS so SystemResolvConf is prohibited.
```
Alternatively, you could delete the first `if` block.

Reply:
I like the switch idea! Implemented locally.